plugins/builder/osbuild.py

```diff
@@ -51,13 +51,17 @@ class Repository:
     def __init__(self, baseurl: str, gpgkey: str = None):
         self.baseurl = baseurl
         self.gpgkey = gpgkey
+        self.rhsm = False
 
     def as_dict(self, arch: str = ""):
         tmp = Template(self.baseurl)
         url = tmp.substitute(arch=arch)
-        res = {"baseurl": url}
+        res = {
+            "baseurl": url,
+            "rhsm": self.rhsm
+        }
         if self.gpgkey:
-            res["gpgkey"] = self.gpgkey
+            res["gpg_key"] = self.gpgkey
```

Comment:
Might need check_gpg = true  here?

Reply:
Good point! This key also propagates to the rpm stage options, if I read the Composer code correctly. I vaguely remember some discussion some time ago that we might not want to gpg verify the packages since we already verified their sha256 while downloading (@teg, correctly me if I am wrong). But I guess that is a Composer policy decision and currently there is no way to separate those and if you have gpg keys we want to verify and thus defaulting to `True` is the right thing here.

Reply:
Ah we might need a little change in the cloudapi then, because I think the cloudapi defaults to false if it's not specified.

Reply:
@Gundersanne I checked the dnf docs, and they also default to false, so maybe that is fine and I hardcode it to `True` here?

Reply:
GPG verification of the packages is sort of silly, I think, as we really should do it at the repo level (and then the packages follow due to their sha's being pinned). However, for backwards compatibility we do the gpg verification in composer, because most repositories are not signed.

For the koji plugin I guess we need to be aware that sometimes nightlies are built from unsigned packages, but we still might want to include the GPG key in the image without actually verifying signatures? I'm not even sure osbuild supports that though :) And I think in practice we don't actually pass the GPG keys to the koji API at the moment, so our images end up key-less (which is a pretty nasty bug if true!)?

Reply:
@teg ok, so what I am hearing is that this might need to become a config option eventually?

Reply:
I think so...